lib/Db/AppointmentConfig.php

```diff
@@ -0,0 +1,185 @@
+ * @method void setAvailability(?string $availability)
+ * @method int|null getStart()
+ * @method void setStart(?int $start)
+ * @method int|null getEnd()
+ * @method void setEnd(?int $end)
+ * @method int getLength()
+ * @method void setLength(int $length)
+ * @method int getIncrement()
+ * @method void setIncrement(int $increment)
+ * @method int getPreparationDuration()
+ * @method void setPreparationDuration(int $prepDuration)
+ * @method int getFollowupDuration()
+ * @method void setFollowupDuration(int $followup)
+ * @method int getTimeBeforeNextSlot()
+ * @method void setTimeBeforeNextSlot(int $buffer)
+ * @method int|null getDailyMax()
+ * @method void setDailyMax(?int $max)
+ */
+class AppointmentConfig extends Entity implements JsonSerializable {
+
+	/** @var string */
+	protected $token;
+
+	/** @var string */
+	protected $name = '';
+
+	/** @var string|null */
+	protected $description;
+
+	/** @var string|null */
+	protected $location;
+
+	/** @var string */
+	protected $visibility;
+
+	/** @var string */
+	protected $userId;
+
+	/** @var string */
+	protected $targetCalendarUri;
+
+	/** @var string|null */
+	protected $calendarFreebusyUris;
+
+	/** @var string|null */
+	protected $availability;
+
+	/** @var int|null */
+	protected $start;
+
+	/** @var int|null */
+	protected $end;
+
+	/** @var int */
+	protected $length;
+
+	/** @var int */
+	protected $increment;
+
+	/** @var int */
+	protected $preparationDuration;
+
+	/** @var int */
+	protected $followupDuration;
+
+	/** @var int */
+	protected $timeBeforeNextSlot;
+
+	/** @var int|null */
+	protected $dailyMax;
+
+	/** @var string */
+	public const VISIBILITY_PUBLIC = 'PUBLIC';
+
+	/** @var string */
+	public const VISIBILITY_PRIVATE = 'PRIVATE';
+
+	/**
+	 * Total length of one slot of the appointment config
+	 * in minutes
+	 *
+	 * @return int  Minutes of Appointment slot length
+	 */
+	public function getTotalLength(): int {
+		return $this->getLength() + $this->getPreparationDuration() + $this->getFollowupDuration();
+	}
+
+	/**
+	 * Principals always have the same format
+	 *
+	 * @return string
+	 */
+	public function getPrincipalUri(): string {
+		return 'principals/users/' . $this->userId;
+	}
+
+	public function getCalendarFreebusyUrisAsArray(): array {
+		return json_decode($this->getCalendarFreebusyUris(), true, 512, JSON_THROW_ON_ERROR);
+	}
+
+	/**
+	 * @param string[] $uris
+	 */
+	public function setCalendarFreeBusyUrisAsArray(array $uris): self {
+		$this->setCalendarFreebusyUris(json_encode($uris));
+
+		return $this;
+	}
+
+	public function jsonSerialize() {
+		return [
+			'id' => $this->id,
+			'token' => $this->getToken(),
+			'name' => $this->getName(),
+			'description' => $this->getDescription(),
+			'location' => $this->getLocation(),
+			'visibility' => $this->getVisibility(),
+			'userId' => $this->getUserId(),
+			'calendarUri' => $this->getTargetCalendarUri(),
+			'calendarFreeBusyUris' => $this->getCalendarFreebusyUris(),
```

Comment:
This serializes to `calendarFreeBusyUris: "[]"` instead of an array. It's a string 
 of an empty array instead of an empty array.

## Example response from a create operation
```json
{
  "status": "success",
  "data": {
    "id": 21,
    "token": "SSJqNdz7NFPE",
    "name": "Test",
    "description": "",
    "location": "",
    "visibility": "PUBLIC",
    "userId": "admin",
    "calendarUri": "/remote.php/dav/calendars/admin/personal/",
    "calendarFreeBusyUris": "[]",
    "availability": "",
    "start": null,
    "end": null,
    "length": 300,
    "increment": 900,
    "preparationDuration": 0,
    "followUpDuration": 0,
    "totalLength": 300,
    "timeBeforeNextSlot": 0,
    "dailyMax": null
  }
}
```

Reply:
Thanks for finding this!

Reply:
This possibly explains why editing configs doesn't work, right?